src/bundles.ts

```diff
@@ -148,37 +158,103 @@ async function fullfillRemoteBundle(
   return remoteBundle;
 }
 
-export async function remoteBundleFactory(
-  baseURL: string,
-  sessionToken: string,
-  files: IFileInfo[],
-  removedFiles: string[] = [],
-  baseDir: string,
-  existingBundleId: string | null = null,
-  maxPayload = MAX_PAYLOAD,
-  source: string,
-): Promise<RemoteBundle | null> {
-  const bundleFactory = prepareRemoteBundle(
-    baseURL,
-    sessionToken,
-    files,
-    removedFiles,
-    existingBundleId,
-    maxPayload,
-    source,
-  );
+interface RemoteBundleFactoryOptions extends PrepareRemoteBundleOptions {
+  baseDir: string;
+}
+
+export async function remoteBundleFactory(options: RemoteBundleFactoryOptions): Promise<RemoteBundle | null> {
   let remoteBundle: RemoteBundle | null = null;
+  const baseOptions = pick(options, ['baseURL', 'sessionToken', 'source', 'baseDir']);
 
+  const bundleFactory = prepareRemoteBundle(omit(options, ['baseDir']));
   for await (const response of bundleFactory) {
     if (response.type === 'error') {
       throw response.error;
     }
 
-    remoteBundle = await fullfillRemoteBundle(baseURL, sessionToken, baseDir, response.value, maxPayload);
+    remoteBundle = await fullfillRemoteBundle({ ...baseOptions, remoteBundle: response.value });
     if (remoteBundle.missingFiles.length) {
       throw new Error(`Failed to upload # files: ${remoteBundle.missingFiles.length}`);
     }
   }
 
   return remoteBundle;
 }
+
+interface CreateBundleFromFoldersOptions extends ConnectionOptions, AnalyzeFoldersOptions {
+  // pass
+}
+
+/**
+ * Get supported filters and test baseURL for correctness and availability
+ *
+ * @param baseURL
+ * @param source
+ * @returns
+ */
+async function getSupportedFiles(baseURL: string, source: string): Promise<SupportedFiles> {
+  emitter.supportedFilesLoaded(null);
+  const resp = await getFilters(baseURL, source);
+  if (resp.type === 'error') {
+    throw resp.error;
+  }
+  const supportedFiles = resp.value;
+  emitter.supportedFilesLoaded(supportedFiles);
+  return supportedFiles;
+}
+
+export interface FileBundle extends RemoteBundle {
+  baseDir: string;
+  supportedFiles: SupportedFiles;
+  fileIgnores: string[];
+}
+
+/**
+ * Creates a remote bundle and returns response from the bundle API
+ *
+ * @param {CreateBundleFromFoldersOptions} options
+ * @returns {Promise<FileBundle | null>}
+ */
+export async function createBundleFromFolders(options: CreateBundleFromFoldersOptions): Promise<FileBundle | null> {
```

Comment:
required: I guess [this one](https://github.com/snyk/code-client/blob/master/src/analysis.ts#L441) can be deleted then?

Reply:
but it was delete.

Reply:
My bad then, must have missed then in github diff. All good then!